cilium-dbg/cmd/encrypt_status.go

```diff
@@ -185,40 +185,176 @@ func maxSequenceNumber() (string, error) {
 	return fmt.Sprintf("0x%x/0xffffffffffffffff", maxSeqNum), nil
 }
 
+// getTunnelDeviceName dynamically determines the tunnel device name from daemon configuration.
+// Returns the device name if tunneling is enabled, empty string if disabled, or error.
+func getTunnelDeviceName() (string, error) {
+	// Check if daemon client is available (not available during testing)
+	if client == nil {
+		return "", fmt.Errorf("daemon client not available")
+	}
+
+	// Get daemon configuration
+	resp, err := client.ConfigGet()
+	if err != nil {
+		return "", fmt.Errorf("failed to get daemon configuration: %w", err)
+	}
+
+	if resp.Status == nil {
+		return "", fmt.Errorf("empty configuration status returned")
+	}
+
+	// Check routing mode to determine if tunneling is enabled
+	// TunnelingEnabled() returns true when routing mode is not "native"
+	routingMode, ok := resp.Status.DaemonConfigurationMap["RoutingMode"]
+	if !ok {
+		return "", fmt.Errorf("routing mode not found in configuration")
+	}
+
+	// If routing mode is native, tunneling is disabled
+	if routingMode == "native" {
+		return "", nil
+	}
+
+	// Get tunnel protocol to determine device name
+	tunnelProtocol, ok := resp.Status.DaemonConfigurationMap["TunnelProtocol"]
+	if !ok {
+		// Default to vxlan if not specified (matches Cilium defaults)
+		tunnelProtocol = "vxlan"
+	}
+
+	// Map tunnel protocol to device name
+	switch tunnelProtocol {
+	case "vxlan":
+		return "cilium_vxlan", nil
+	case "geneve":
+		return "cilium_geneve", nil
+	default:
+		return "", fmt.Errorf("unsupported tunnel protocol: %v", tunnelProtocol)
+	}
+}
```

Comment:
I see [CI fail](https://github.com/cilium/cilium/actions/runs/17727775368/job/50372199602) with:
```
failed to fetch encryption status from cilium-5zn7v: command failed (pod=kube-system/cilium-5zn7v, container=cilium-agent): command terminated with exit code 1: "Error: Cannot get daemon encryption status: error getting IPsec decryption interfaces: failed to check tunnel decryption interface: failed to get tunnel device cilium_vxlan: Link not found\n"
```
so I suspect there's an issue with this function.